src/openrct2-ui/windows/Park.cpp

```diff
@@ -1479,40 +1491,263 @@ static void window_park_objective_paint(rct_window* w, rct_drawpixelinfo* dpi)
         {
             rideTypeString = RideTypeDescriptors[rideTypeId].Naming.Name;
         }
-        ft.Add<rct_string_id>(rideTypeString);
+        ft->Add<rct_string_id>(rideTypeString);
     }
     else
     {
-        ft.Add<uint16_t>(gScenarioObjective.NumGuests);
-        ft.Add<int16_t>(date_get_total_months(MONTH_OCTOBER, gScenarioObjective.Year));
+        ft->Add<uint16_t>(gScenarioObjective.NumGuests);
+        ft->Add<int16_t>(date_get_total_months(MONTH_OCTOBER, gScenarioObjective.Year));
         if (gScenarioObjective.Type == OBJECTIVE_FINISH_5_ROLLERCOASTERS)
-            ft.Add<uint16_t>(gScenarioObjective.MinimumExcitement);
+            ft->Add<uint16_t>(gScenarioObjective.MinimumExcitement);
         else
-            ft.Add<money32>(gScenarioObjective.Currency);
+            ft->Add<money32>(gScenarioObjective.Currency);
     }
 
-    screenCoords.y += gfx_draw_string_left_wrapped(
-        dpi, ft.Data(), screenCoords, 221, ObjectiveNames[gScenarioObjective.Type], COLOUR_BLACK);
-    screenCoords.y += 5;
+    screenCoords->y += gfx_draw_string_left_wrapped(
+        dpi, ft->Data(), *screenCoords, 221, ObjectiveNames[gScenarioObjective.Type], COLOUR_BLACK);
+    screenCoords->y += 5;
+}
 
-    // Objective outcome
-    if (gScenarioCompletedCompanyValue != MONEY32_UNDEFINED)
+static void window_park_objective_paint_scenario_progress(
+    rct_window* w, rct_drawpixelinfo* dpi, Formatter* ft, ScreenCoordsXY* screenCoords)
+{
+    if (gScenarioObjective.GetCurrentObjectiveStatus() == ObjectiveStatus::Undecided)
     {
-        if (gScenarioCompletedCompanyValue == COMPANY_VALUE_ON_FAILED_OBJECTIVE)
+        gfx_draw_string_left(dpi, STR_OBJECTIVE_PROGRESS_LABEL, nullptr, COLOUR_BLACK, *screenCoords);
+        screenCoords->y += LIST_ROW_HEIGHT;
+
+        // Progress
+        int32_t barColour = COLOUR_BRIGHT_GREEN;
+
+        switch (gScenarioObjective.Type)
+        {
+            case OBJECTIVE_NONE:
+            case OBJECTIVE_HAVE_FUN:
+            case OBJECTIVE_BUILD_THE_BEST:
+                screenCoords->y += gfx_draw_string_left_wrapped(
+                    dpi, nullptr, *screenCoords, 221, STR_OBJECTIVE_PROGRESS_NONE, COLOUR_BLACK);
+                break;
+            case OBJECTIVE_GUESTS_BY:
+            case OBJECTIVE_GUESTS_AND_RATING:
+            {
+                float guestsInParkPercentage
+                    = (static_cast<float>(gNumGuestsInPark) / static_cast<float>(gScenarioObjective.NumGuests));
+                float parkRatingPercentage;
```

Comment:
Move this line to just before the `if (gScenarioObjective.Type == OBJECTIVE_GUESTS_BY)`

You can even initialize it to `static_cast<float>(gParkRating)` and then just divide it inside the `if` or `else`